source/base/function_parser.cc

```diff
@@ -176,6 +179,27 @@ namespace internal
   {
     return erfc(value);
   }
+
+  // returns a random value in the range [0,1]
+  double mu_rand_seed(double seed)
+  {
+    std::mutex mtx;
+    mtx.lock();
+    static boost::random::mt19937 rng((unsigned int)seed);
+    mtx.unlock();
+    boost::random::uniform_real_distribution<> uniform_distribution(0,1);
+    return uniform_distribution(rng);
+  }
+
+  // returns a random value in the range [0,1]
+  double mu_rand()
+  {
+    boost::random::mt19937 rng;
+    rng.seed(static_cast<unsigned long>(std::time(0)));
+    boost::random::uniform_real_distribution<> uniform_distribution(0,1);
```

Comment:
if you make the `rng` static, then you don't need to seed it every time.

This function is also missing the mutex.